### DISCUSSION TITLE
Running the installation command gives me "No project was found at the path" messages

### DISCUSSION THREAD

**Discussion by mrlucmorin** [AvaloniaUI/Avalonia.Templates#284] [Q&A]
Hi,

As stated in the title, when I run the command to install the Avalonia templates, some of them issue messages such as:

> The following templates might not work because their constraints are not met:
> Avalonia TemplatedControl (avalonia.templatedcontrol) C#(identity: Avalonia.TemplatedControl)
>    Project capabiltities: No project was found at the path: C:\Users\Luc. This template can only be created inside the project.

The command I used to install is:

> dotnet new install Avalonia.Templates

Some of the templates seem to get installed. I can see them with:

`dotnet new list`

> Avalonia .NET App                             avalonia.app                project
> Avalonia .NET MVVM App                        avalonia.mvvm               project
> Avalonia Cross Platform Application           avalonia.xplat              solution
> 

What I seem to understand is that I'm running the command outside of a project folder, but nothing in the documentation specifies that templates should be installed only at the "project level".

OS: Windows 10
dotnet: 9.0.101
Terminal: Powershell 7.4.6 

Am I missing a step here?

Thanks

**Comment by maxkatz6** [comment]
> but nothing in the documentation specifies that templates should be installed only at the "project level".

It's not a requirement. You should be able to install templates in any folder.

> The following templates might not work because their constraints are not met

It's likely a bug in .NET SDK tooling. 
SDK supports project capabilities constraint on templates - https://github.com/dotnet/templating/wiki/Constraints#project-capabilities
Which [we use](https://github.com/AvaloniaUI/avalonia-dotnet-templates/pull/279/files) for file templates, such as "user control template", which only make sense inside of Avalonia projects.

For some reason, SDK false positives with this constraint on a plain "dotnet new install"

**Reply by mrlucmorin** [reply]
Hi,

Thanks for the reply. It's nice to know I'm not the first one to see/report this.

So basically I should ignore those messages? I mean, I can still see those templates in Rider, so they have to be "installed" somewhere for Rider to see them, right? 

Cheers!

EDIT: BTW, I had seen those lines of code in the template.json for those same for which I was seeing the "error" message, but I'm too green yet to properly interpret what that "constraints" section actually does. Would you be so kind as to point me to some docs about the template.json file's overall functionality?

Thanks

**Comment by ahinterl** [comment]
I was actually installing the Avalonia templates today (current JetBrains Rider version and .NET 10), and I'm seeing exactly what the author describes.

Am I safe to ignore the "The following templates might not work because their constraints are not met" warning?
